### ISSUE TITLE
Train on multiple graphs

### ISSUE THREAD

**Issue by vijay2411** [rusty1s/pytorch_geometric#1741]
## ❓ Training by considering multiple Graphs from a dataset. Confirmation!!

First of all, A very very thanks for creating this library. It works wonderfully, and saves a lot of time.

Now,
Suppose we have the cora dataset. I want to train a graphSAGE model but considering various graphs simultaneously!
I mean, I want to train by totaling the loss from each graph in the cora dataset i.e. cora[0], cora[1] ... cora[length of cora dataset -1].
So, in one iteration I can just sum the loss += mseLoss(cora[i].Y - GraphSage(cora[i])) for i in cora.size(), and can apply backprop, so it should work, right?

P.S. How to create a Manual Dataset. I can create Data from torch_geometric.data but I couldn't see any manual method to create a dataset other than implementing the class InMemoryDataset.
Thank You.

**Comment by rusty1s** [comment]
Yes, you can do that. Alternatively, you can use `torch_geometric.data.DataLoader` to train in mini-batches:
```python
loader = DataLoader([cora] * N, batch_size=...)

for data in loader:
     mse_loss(...)
```
